ambry-server/src/integration-test/java/com.github.ambry.server/ServerStorePreFetchTest.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+
+@RunWith(Parameterized.class)
+public class ServerStorePreFetchTest {
+  private static Properties routerProps;
+  private static MockNotificationSystem notificationSystem;
+  private static MockCluster plaintextCluster;
+  private final boolean testEncryption;
+
+  @BeforeClass
+  public static void initializeTests() throws Exception {
+    routerProps = new Properties();
+    routerProps.setProperty("kms.default.container.key", TestUtils.getRandomKey(32));
+    notificationSystem = new MockNotificationSystem(9);
+    plaintextCluster = new MockCluster(notificationSystem, new Properties(), false, true, SystemTime.getInstance());
+    plaintextCluster.startServers();
+  }
+
+  /**
+   * Running for both regular and encrypted blobs
+   * @return an array with both {@code false} and {@code true}.
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{true}});
+  }
+
+  public ServerStorePreFetchTest(boolean testEncryption) throws Exception {
+    this.testEncryption = testEncryption;
+  }
+
+  @AfterClass
+  public static void cleanup() throws IOException {
+    long start = System.currentTimeMillis();
+    // cleanup appears to hang sometimes. And, it sometimes takes a long time. Printing some info until cleanup is fast
+    // and reliable.
+    System.out.println("About to invoke cluster.cleanup()");
+    if (plaintextCluster != null) {
+      plaintextCluster.cleanup();
+    }
+    System.out.println("cluster.cleanup() took " + (System.currentTimeMillis() - start) + " ms.");
+  }
+
+  @Test
+  public void startStopTest() throws IOException, InstantiationException, URISyntaxException, GeneralSecurityException {
+  }
+
+  @Test
+  public void endToEndTest()
```

Comment:
our integration tests are quite slow :( Adding these many will be bad. Instead of introducing a class, maybe we can 
1. Make a single server do prefetch (and know which server so that we can explicitly get from it)
2. In `ServerPlaintextTest`, set the prefetch to true in `endToEndTest()` and leave is as `false` in the other two.

Will chat with you offline to understand the options

Reply:
Use a different server whose prefetch is enabled for eneToEndTest() in both plaintext and ssl test.